File: app/src/main/java/com/example/readative/room/entity/Bookmark.java
package com.example.readative.room.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Bookmark {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "bookmark_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "book_owner_id")
    public long bookOwnerId;

    @NonNull
    @ColumnInfo(name = "bookmark_name")
    public String name;
    @NonNull
    @ColumnInfo(name = "bookmark_page")
    public int page;
}

File: app/src/main/java/com/example/readative/persistance/entities/Bookmark.kt
package com.example.readative.persistance.entities

import androidx.room.*

@Entity(
    tableName = "bookmarks",
    indices = [
        Index(value = ["page"], unique = true)
    ],
    foreignKeys = [
        ForeignKey(
            entity = Book::class,
            parentColumns = ["id"],
            childColumns = ["book_id"],
            onUpdate = ForeignKey.CASCADE,
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Bookmark(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "book_id") override val bookId: Long,
    @ColumnInfo(name = "name") val name: String,
    @ColumnInfo(name = "page") val page: Int
) : ReadativeEntity, BookIdEntity
